Implement lists:split/2

petermm opened the pull request:
As implemented in lists: module in OTP https://github.com/erlang/otp/blob/0e66b88db447954e1de70ec1fa0822d3a7e6133d/lib/stdlib/src/lists.erl#L1808

Unsure where the line goes for functions to be implemented by estdlib, so feel free to close PR.

Use case: 
Using https://github.com/discord/limited_queue/ on top of the erlang queue: module (copy pasted), and the only missing lists function needed was the split/2 - so would be nice to have it out of the box.

These changes are made under both the "Apache 2.0" and the "GNU Lesser General
Public License 2.1 or later" license terms (dual license).

SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later

fadushin commented on libs/estdlib/src/lists.erl:
```diff
@@ -502,6 +503,30 @@ sort(List) when is_list(List) ->
 sort(Fun, List) when is_function(Fun), is_list(List) ->
     quick_sort(Fun, List).
 
+-spec split(N, List1) -> {List2, List3} when
```
Could you please add API docs?

bettio reviewed (commented):
Just to understand, is it your lists:split implementation or the one taken from Erlang?

petermm commented:
> Just to understand, is it your lists:split implementation or the one taken from Erlang?

100% from Erlang.

bettio commented:
> > Just to understand, is it your lists:split implementation or the one taken from Erlang?
> 
> 100% from Erlang.

So if you took that function from Erlang (and it is not your code) please update the module copyright in order to mention the original copyright, also mention the source of that code in your commit.

petermm commented:
> > > Just to understand, is it your lists:split implementation or the one taken from Erlang?
> > 
> > 
> > 100% from Erlang.
> 
> So if you took that function from Erlang (and it is not your code) please update the module copyright in order to mention the original copyright, also mention the source of that code in your commit.

Added, couldn't find an example elsewhere in codebase- so do let me know if further changes are required..
